[fix bug 1411077] Whatsnew page for Firefox 57 Quantum

craigcook opened the pull request:
## Description
Firefox 57 will have a custom whatsnew page with a few variations. One version will be a letter from Chris Beard translated in 8 primary languages, and the other is a minimal baseline for all other locales.

## Issue / Bugzilla link
[Baseline page for all locales](https://bugzilla.mozilla.org/show_bug.cgi?id=1411077)
[Letter variant for key locales](https://bugzilla.mozilla.org/show_bug.cgi?id=1411076)

## Testing
[Demo](https://bedrock-demo-craigcook.us-west.moz.works/firefox/57.0/whatsnew/)

craigcook commented:
@peiying2  I'll be adding locale-specific templates for the 8 primary locales that will show the letter and they'll be excluded from the regular l10n process so we're only concerned with `whatsnew-57.html` for now (lang file is `firefox/whatsnew_57`).

The baseline page has a total of five new strings, all short and simple, so hopefully it won't present too much of a burden for localizers. Let me know if you spot any issues. Thanks!

peiying2 commented:
@craigcook the file is prepared. We need to know the demo URL for this page so localizers can verify their work in context.

flodolo commented on media/css/firefox/whatsnew/whatsnew-57.scss:
```diff
@@ -0,0 +1,240 @@
+        .tagline {
+            margin: 20px 0 40px;
+
+            br {
```
Need to remove this (CC @MikkCZ)
https://bedrock-demo-craigcook.us-west.moz.works/cs/firefox/57.0/whatsnew/

flodolo replied:
For reference, this is really bad. You would end up removing a perfectly fine `<br>`, and breaking the sentence (e.g. missing a space after a period or comma).

craigcook commented on bedrock/firefox/templates/firefox/new/quantum/base.html:
```diff
@@ -45,7 +45,7 @@
     <div class="content">
       <div class="header-logos">
-        <h2><a class="firefox" href="{{ url('firefox') }}">{{ high_res_img('logos/firefox/logo-quantum-wordmark-white.png', {'alt': 'Firefox', 'width': '133', 'height': '51'}) }}</a></h2>
+        <h2><a class="firefox" href="{{ url('firefox') }}">{{ high_res_img('logos/firefox/logo-quantum-wordmark-white.png', {'alt': 'Firefox', 'width': '133', 'height': '50'}) }}</a></h2>
```
I needed a slightly bigger version of this image but didn't want to add yet another dupe logo so I updated the original but had to tweak the size here. This was the only other place it was used so far, but if other new pages also rely on this asset they may need to adjust sizing.

bigdickbrinson67 replied:
Yes

craigcook commented:
All locales have signed off on the letter translations so this is ready for code review.

alexgibson commented on media/css/firefox/whatsnew/whatsnew-57.scss:
```diff
@@ -0,0 +1,236 @@
+        }
+
+        @media (min-height: 800px) {
+            min-height: 60vh;
```
On my laptop this value feels kinda big, with most of the letter off below screen. `40vh` feels about right though. Could this do with an additional check for a really large width?

craigcook commented on bedrock/firefox/views.py:
```diff
@@ -340,20 +340,29 @@ def show_54_whatsnew(version):
 
     if oldversion:
-        return version >= v56 and oldversion < v56
+        return version >= v57 and oldversion < v57
```
I just copied the old tests. I fully expected to get this wrong.
![noidea-jingle](https://user-images.githubusercontent.com/205591/32614516-4e8900fc-c522-11e7-84cd-5ac141b4ed17.gif)

craigcook commented:
Updated! I was clearly overthinking the view and tests. We don't need to do anything special for oldversion and the v56 logic should still apply for v56, so I've reverted that stuff and simplified the 57 bits.

craigcook commented:
Also of note, flod has expressed concerns about locale-specific templates without an active lang file not loading properly in production mode, so we should test that thoroughly (I guess there were issues with this before but maybe that was a bug that got fixed). It all works as it should for me locally with DEV=False but I don't know if I fully trust that on a local environment. Maybe we can set a demo server to production mode to verify?
